### PR TITLE
Arnej/config class should not be public

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@bratseth please review
@baldersheim @hmusum @toregge FYI

### CODE REVIEW

**Review by bratseth — APPROVED** [review]
👍

File: document/src/vespa/document/config/doctypecfg.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * DocumenttypesConfig and DocumenttypesConfigBuilder
+ * (globally visible).
+ **/
+
+namespace document::config::internal {
+class InternalDocumenttypesType;
+}
+
+using DocumenttypesConfigBuilder = document::config::internal::InternalDocumenttypesType;
+using DocumenttypesConfig = const document::config::internal::InternalDocumenttypesType;
```
Comment (toregge) [inline document/src/vespa/document/config/doctypecfg.h:16]: Consider moving the using statements to the `document::config` namespace, i.e. the same namespace that is used when the generated `config-documenttypes.h` is included.

Consider renaming the header file to something containing fwd, e.g. as `iosfwd` contains forward declarations for `iostream`.

**Review by toregge — APPROVED** [review]

**Comment by arnej27959** [comment]
> Consider moving the using statements to the document::config namespace

I tried that first, but it felt like more work for less consistent and less readable code.

**Comment by toregge** [comment]
> I tried that first, but it felt like more work for less consistent and less readable code.

But that means that `config-documenttypes.h` won't work as a drop-in replacement for `documenttypes_config_fwd.h`, you might need both if you depend on the using being in the global namespace and you need the actual config definition.

**Pull request merged**